skbio/diversity/alpha/_base.py

```diff
@@ -8,14 +8,114 @@
 
 from __future__ import absolute_import, division, print_function
 
+from functools import partial
+
 import numpy as np
+import pandas as pd
 from scipy.special import gammaln
 from scipy.optimize import fmin_powell, minimize_scalar
 
+import skbio
 from skbio.stats import subsample_counts
 from skbio.util._decorator import experimental
 from skbio.diversity._base import (_validate_counts_vector,
-                                   _validate_otu_ids_and_tree)
+                                   _validate_otu_ids_and_tree,
+                                   _vectorize_counts_and_tree)
+
+
+# need to figure out how to better factor this. as it stands, it's a big
+# duplication of .__init__.__all__
```

Comment:
There should be a private ``dict`` which maps strings to functions, and then a public function, ``skbio.diversity.get_alpha_diversity_metrics()`` that returns the keys of the private ``dict``.